app/internal/page/your_details.go

```diff
@@ -86,21 +96,51 @@ func (d *yourDetailsForm) Validate() map[string]string {
 	if d.FirstNames == "" {
 		errors["first-names"] = "enterFirstNames"
 	}
+	if len(d.FirstNames) > 53 {
+		errors["first-names"] = "firstNamesTooLong"
+	}
```

Comment:
Probably not something we can resolve on this PR but do we have the research/reasoning on these specific limits on name lengths? Feels like a hangover from needing to print to a paper form.

Reply:
I'm just taking them from what was in the spreadsheet shared from Make